### ISSUE TITLE
[Bug]: Integration of Rich Notification on iOS errors out EAS Build

### ISSUE THREAD

**Issue by mvolonnino** [braze-inc/braze-expo-plugin#30]
### Which Platforms?

iOS

### Which React Native Version?

0.73.6

### Which @braze/expo-plugin version?

^2.1.0

### Which @braze/react-native-sdk SDK version?

^8.3.0

### Repro Rate

100%

### Steps To Reproduce

Run a release build through EAS and it fails to build each time while `enableBrazeIosRichPush` is set to true in the `app.config.ts`. Removing that flag in the braze plugin allows for the build to succeed.

### Expected Behavior

Build should not fail, and rich notification support should be enabled for iOS & braze push notifications

### Actual Incorrect Behavior

App fails to build during the _fastlane_ setup through EAS.

### Verbose Logs

```shell
➜ Explicit dependency on target 'BrazeExpoRichPush' in project 'Volo'
        ➜ Implicit dependency on target 'BrazeUI' in project 'Pods' via options '-framework BrazeUI' in build setting 'OTHER_LDFLAGS'
        ➜ Implicit dependency on target 'ExpoAdapterBraze' in project 'Pods' via options '-framework ExpoAdapterBraze' in build setting 'OTHER_LDFLAGS'
        ➜ Implicit dependency on target 'braze-react-native-sdk' in project 'Pods' via options '-framework braze_react_native_sdk' in build setting 'OTHER_LDFLAGS'
        ➜ Explicit dependency on target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeLocation' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeUI' in project 'Pods'
        ➜ Explicit dependency on target 'ExpoAdapterBraze' in project 'Pods'
        ➜ Explicit dependency on target 'braze-react-native-sdk' in project 'Pods'
    Target 'ExpoAdapterBraze' in project 'Pods'
        ➜ Explicit dependency on target 'braze-react-native-sdk' in project 'Pods'
    Target 'braze-react-native-sdk' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeLocation' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeUI' in project 'Pods'
    Target 'BrazeUI' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeUI-BrazeUI' in project 'Pods'
    Target 'BrazeUI-BrazeUI' in project 'Pods' (no dependencies)
    Target 'BrazeLocation' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeLocation-BrazeLocation' in project 'Pods'
    Target 'BrazeLocation-BrazeLocation' in project 'Pods' (no dependencies)
    Target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeKit-BrazeKit' in project 'Pods'
    Target 'BrazeKit-BrazeKit' in project 'Pods' (no dependencies)
    Target 'BrazeExpoRichPush' in project 'Volo'
        ➜ Implicit dependency on target 'Pods-BrazeExpoRichPush' in project 'Pods' via file 'Pods_BrazeExpoRichPush.framework' in build phase 'Link Binary'
    Target 'Pods-BrazeExpoRichPush' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeNotificationService' in project 'Pods'
    Target 'BrazeNotificationService' in project 'Pods' (no dependencies)
/Users/expo/workingdir/build/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 11.0, but the range of supported deployment target versions is 12.0 to 17.2.99. (in target 'BrazeUI-BrazeUI' from project 'Pods')
/Users/expo/workingdir/build/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 11.0, but the range of supported deployment target versions is 12.0 to 17.2.99. (in target 'BrazeLocation-BrazeLocation' from project 'Pods')
/Users/expo/workingdir/build/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 11.0, but the range of supported deployment target versions is 12.0 to 17.2.99. (in target 'BrazeKit-BrazeKit' from project 'Pods')
/Users/expo/workingdir/build/ios/Volo.xcodeproj: error: Signing for "BrazeExpoRichPush" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'BrazeExpoRichPush' from project 'Volo')
```

### Additional Information

@jerielng - now this is awesome! One question though as I am running into build errors when updating everything to expo sdk50 & updating to the latest 2.1.0 version.

Getting this build error in the eas-build process:

> Starting from Xcode 14, resource bundles are signed by default, which requires setting the development team for each resource bundle target.

and from inspecting the xcode logs, im seeing this issue:
```
        ➜ Explicit dependency on target 'BrazeExpoRichPush' in project 'Volo'
        ➜ Implicit dependency on target 'BrazeUI' in project 'Pods' via options '-framework BrazeUI' in build setting 'OTHER_LDFLAGS'
        ➜ Implicit dependency on target 'ExpoAdapterBraze' in project 'Pods' via options '-framework ExpoAdapterBraze' in build setting 'OTHER_LDFLAGS'
        ➜ Implicit dependency on target 'braze-react-native-sdk' in project 'Pods' via options '-framework braze_react_native_sdk' in build setting 'OTHER_LDFLAGS'
        ➜ Explicit dependency on target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeLocation' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeUI' in project 'Pods'
        ➜ Explicit dependency on target 'ExpoAdapterBraze' in project 'Pods'
        ➜ Explicit dependency on target 'braze-react-native-sdk' in project 'Pods'
    Target 'ExpoAdapterBraze' in project 'Pods'
        ➜ Explicit dependency on target 'braze-react-native-sdk' in project 'Pods'
    Target 'braze-react-native-sdk' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeLocation' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeUI' in project 'Pods'
    Target 'BrazeUI' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeUI-BrazeUI' in project 'Pods'
    Target 'BrazeUI-BrazeUI' in project 'Pods' (no dependencies)
    Target 'BrazeLocation' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeLocation-BrazeLocation' in project 'Pods'
    Target 'BrazeLocation-BrazeLocation' in project 'Pods' (no dependencies)
    Target 'BrazeKit' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeKit-BrazeKit' in project 'Pods'
    Target 'BrazeKit-BrazeKit' in project 'Pods' (no dependencies)
    Target 'BrazeExpoRichPush' in project 'Volo'
        ➜ Implicit dependency on target 'Pods-BrazeExpoRichPush' in project 'Pods' via file 'Pods_BrazeExpoRichPush.framework' in build phase 'Link Binary'
    Target 'Pods-BrazeExpoRichPush' in project 'Pods'
        ➜ Explicit dependency on target 'BrazeNotificationService' in project 'Pods'
    Target 'BrazeNotificationService' in project 'Pods' (no dependencies)
/Users/expo/workingdir/build/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 11.0, but the range of supported deployment target versions is 12.0 to 17.2.99. (in target 'BrazeLocation-BrazeLocation' from project 'Pods')
/Users/expo/workingdir/build/ios/Volo.xcodeproj: error: Signing for "BrazeExpoRichPush" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'BrazeExpoRichPush' from project 'Volo')
/Users/expo/workingdir/build/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 11.0, but the range of supported deployment target versions is 12.0 to 17.2.99. (in target 'BrazeUI-BrazeUI' from project 'Pods')
/Users/expo/workingdir/build/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 11.0, but the range of supported deployment target versions is 12.0 to 17.2.99. (in target 'BrazeKit-BrazeKit' from project 'Pods')
```

Looks like its failing on the signing for the Rich Push notifications. I could not find anything on this for the expo config plugin - so wondering how to handle this here and allow the rich push notifications in the latest 2.1.0!

**Comment by jerielng** [comment]
Hey @mvolonnino, we've released version 2.1.1 of this Expo plugin. Could you verify if this resolves your issue? Thanks!

**Comment by mvolonnino** [comment]
@jerielng - Awesome! Im a little backed up, but will get to this within the next week 🔥

**Comment by jerielng** [comment]
@mvolonnino We've gotten word that 2.1.1 may not fully address some of these code signing issues, so we're taking a deeper look into it. Sorry for the back and forth, but we'll keep you updated as much as we can when we have a more holistic fix. Thanks!

**Comment by krazykriskomar** [comment]
Hi there @jerielng. Kris Komar from The Krazy Coupon lady here. I'm lead architect here. Just letting you know we are experiencing an issue with rich pushes not working in iOS in Expo and it looks like this particular issue might be the culprit. Specifically, when we build with enableBrazeIosRichPush = true, the build fails. Do you need any other data? I'm happy to provide it. How's the fix for this going? Do you know when there might be a new version to remedy this? Would you like any other assistance in the matter? 
Thanks in advance!

**Comment by jerielng** [comment]
Hi @krazykriskomar, we currently have the fix on a preview repo [here](https://github.com/braze-inc/braze-expo-plugin-eas-preview) if you would like to temporarily point your project to it. If this fix fully resolves the issue for you, let us know!

**Comment by krazykriskomar** [comment]
> Hi @krazykriskomar, we currently have the fix on a preview repo [here](https://github.com/braze-inc/braze-expo-plugin-eas-preview) if you would like to temporarily point your project to it. If this fix fully resolves the issue for you, let us know!

@jerielng I would love to test it out! :D How do I consume this? Is it on npmjs or do I need to manually download the source?

**Comment by jerielng** [comment]
@krazykriskomar You can update the pointer in your `package.json`. Instead of using a version number, you can either clone the repo locally and refer to it like so:
```
"@braze/expo-plugin": "file:<relative-path-to>/braze-expo-plugin-eas-preview",
```
or you can use the direct Git link (If you are not using SSH, there are some examples of additional configurations [here](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#urls-as-dependencies)):
```
"@braze/expo-plugin": "git+ssh://github.com/braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",
```

**Comment by krazykriskomar** [comment]
> @krazykriskomar You can update the pointer in your `package.json`. Instead of using a version number, you can either clone the repo locally and refer to it like so:
> 
> ```
> "@braze/expo-plugin": "file:<relative-path-to>/braze-expo-plugin-eas-preview",
> ```
> 
> or you can use the direct Git link (If you are not using SSH, there are some examples of additional configurations [here](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#urls-as-dependencies)):
> 
> ```
> "@braze/expo-plugin": "git+ssh://github.com/braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",
> ```

You're a lifesaver, thank you! :D

**Comment by krazykriskomar** [comment]
@jerielng it's giving me an error on expo install 
<img width="919" alt="image" src="https://github.com/braze-inc/braze-expo-plugin/assets/87501600/dbcce09e-2537-4670-bb9f-ced284e28c2c">
Do I perhaps need to modify my app.config.js? It currently looks like this: 
<img width="497" alt="image" src="https://github.com/braze-inc/braze-expo-plugin/assets/87501600/e4c2c049-0df1-4c7b-93f3-f3d3f93f450d">
And I changed package.json as such:
<img width="715" alt="image" src="https://github.com/braze-inc/braze-expo-plugin/assets/87501600/f2c00614-fad5-4cf5-b582-ba6a69e64df5">

**Comment by jerielng** [comment]
@krazykriskomar Looks like there was a typo in that link, sorry. It should be this if you're using SSH:
```
"@braze/expo-plugin": "git+ssh://git@github.com:braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",
```

**Comment by krazykriskomar** [comment]
@jerielng 
Since this isn't a published package, it doesn't contain a /build folder, so it will always fail. Would you like me to add a "prepare" script to the package.json and if that works I can put in a PR here?

**Comment by krazykriskomar** [comment]
> @krazykriskomar Looks like there was a typo in that link, sorry. It should be this if you're using SSH:
> 
> ```
> "@braze/expo-plugin": "git+ssh://git@github.com:braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",
> ```

@jerielng thank you for your help! When were you guys planning to push this to prod/npm?

**Comment by jerielng** [comment]
> > @krazykriskomar Looks like there was a typo in that link, sorry. It should be this if you're using SSH:
> > ```
> > "@braze/expo-plugin": "git+ssh://git@github.com:braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",
> > ```
> 
> @jerielng thank you for your help! When were you guys planning to push this to prod/npm?

We're hoping to get this pushed to our main repo shortly. We'll keep you updated on a timeline for it. Thanks!

**Comment by krazykriskomar** [comment]
@jerielng by chance are you still experiencing this error? 
`error: Signing for "KrazyCouponLady" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'KrazyCouponLady' from project 'KrazyCouponLady')`
I get this only when building in release mode. Here's the raw command that Expo is running:
`xcodebuild -workspace ./KrazyCouponLady.xcworkspace -scheme KrazyCouponLady -configuration Release -destination 'generic/platform=iOS' -archivePath /Users/expo/Library/Developer/Xcode/Archives/2024-06-06/KrazyCouponLady\ 2024-06-06\ 06.16.44.xcarchive archive | tee /Users/expo/workingdir/logs/KrazyCouponLady-KrazyCouponLady.log`

**Comment by jerielng** [comment]
Hey @krazykriskomar Do you happen to have a minimum reproducible sample of this issue that you can share with our support team at support@braze.com? We're not able to reproduce this from our end. The automatic signing configuration should already be enabled for each of the rich push and push stories extensions as part of the script that generates these targets.

**Comment by krazykriskomar** [comment]
@jerielng I'll work on creating a reproducible version. 
Question for you: I downloaded a copy of the preview repo last Wednesday and that's the code I'm working on. Have you made any updates that I might need to update since then?

**Comment by jerielng** [comment]
Hey @krazykriskomar, we haven't made any updates to that repo.

One other troubleshooting step I would also recommend is checking that you've added the app extension bundle IDs to your Expo project configurations so that when you run the `eas build` step, you are able to generate the provisioning profiles for them. There's a couple of ways you can do this depending on how you typically manage code signing in your project. Here's one such [example](https://docs.expo.dev/build-reference/app-extensions/). Or you can also add it to your [credentials.json](https://docs.expo.dev/app-signing/local-credentials/#multi-target-project) if that's where you normally keep your code signing properties.

**Comment by krazykriskomar** [comment]
> Hey @krazykriskomar, we haven't made any updates to that repo.
> 
> One other troubleshooting step I would also recommend is checking that you've added the app extension bundle IDs to your Expo project configurations so that when you run the `eas build` step, you are able to generate the provisioning profiles for them. There's a couple of ways you can do this depending on how you typically manage code signing in your project. Here's one such [example](https://docs.expo.dev/build-reference/app-extensions/). Or you can also add it to your [credentials.json](https://docs.expo.dev/app-signing/local-credentials/#multi-target-project) if that's where you normally keep your code signing properties.

@jerielng if there is extra setup required to use this plugin, perhaps I missed it. Where is this setup/config in the documentation?

**Comment by jerielng** [comment]
@krazykriskomar If you are using EAS CLI, you'll need to take some additional steps to ensure that:
* you have added the bundle IDs for those app extension targets to your EAS configuration (This can be done with multiple approaches as described above, so you would have to choose the best approach that fits your project).
* when you run `eas build`, run it with the `EXPO_NO_CAPABILITY_SYNC=1` flag (There is an ongoing issue [here](https://github.com/expo/eas-cli/issues/1069) where EAS automatically removes the capabilities attached to a provisioning profile. So you'll need to use this flag to ensure that that automation doesn't occur.)

We don't currently have these documented as they can be specific based on the project that uses EAS, but we'll look into getting it updated on our public docs [here](https://www.braze.com/docs/developer_guide/platform_integration_guides/react_native/push_notifications/#enabling-rich-push-notifications-for-ios) once we have made a public release of the new changes.

**Comment by krazykriskomar** [comment]
@jerielng figured it out! I had to run an expo build locally. We use CI for our builds and CI cannot create provisioning certs.

**Comment by YentingChen** [comment]
Hi @jerielng 
I tried to point to the repository by updating the plugin to 

`"@braze/expo-plugin": "git+ssh://git@github.com:braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",`

but I still got the error when executing `expo rebuild`

![Screenshot 2024-06-11 at 16 01 35](https://github.com/braze-inc/braze-expo-plugin/assets/14238145/bec7d8a6-8230-4988-ae7b-6a956106b3e2)

Could you please help? Thank you.

**Comment by jerielng** [comment]
> Hi @jerielng I tried to point to the repository by updating the plugin to
> 
> `"@braze/expo-plugin": "git+ssh://git@github.com:braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",`
> 
> but I still got the error when executing `expo rebuild`
> 
> ![Screenshot 2024-06-11 at 16 01 35](https://private-user-images.githubusercontent.com/14238145/338638690-bec7d8a6-8230-4988-ae7b-6a956106b3e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMjg1NjEsIm5iZiI6MTcxODEyODI2MSwicGF0aCI6Ii8xNDIzODE0NS8zMzg2Mzg2OTAtYmVjN2Q4YTYtODIzMC00OTg4LWFlN2ItNmE5NTYxMDZiM2UyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDE3NTEwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5Nzc4NWJkMzM1MGFkMjkzMzdmNDQyMzFhMDUzODBlNGJmNjAwODQ3MDY3Yjg5ZGY2ZDUxYjVkMTZjY2QwNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZbeIdJkdrI4grCL0mZSQTDa3GrUruwV1uHkJ-eu0oyI)
> 
> The build folder does not seem to have been generated. ![Screenshot 2024-06-11 at 16 22 14](https://private-user-images.githubusercontent.com/14238145/338647193-88e3fb16-f789-4501-9570-a4e5d757f9b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMjg1NjEsIm5iZiI6MTcxODEyODI2MSwicGF0aCI6Ii8xNDIzODE0NS8zMzg2NDcxOTMtODhlM2ZiMTYtZjc4OS00NTAxLTk1NzAtYTRlNWQ3NTdmOWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDE3NTEwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3Y2Y2Y2I1MDFiNDA3MDlkNmYyNWI4ZDhkODZhZGUwMTNkMzQxNDk3YmFiMjRlNDFmOTZkM2RjZTQ5ZGVlYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1-pdqcA-pLn6ZiMeZUXV_whP0BW1Pjqg176HgMe0uOg)
> 
> Could you please help? Thank you.

Hey @YentingChen, it looks like the plugin files aren't being synced. Could you alternatively try cloning the repo and pointing to a local build using the `file:` syntax?

We're hoping to get this published soon after it goes through some internal review, so you should be able to point directly to our NPM instance once the new version has been released.

**Comment by krazykriskomar** [comment]
> Hi @jerielng I tried to point to the repository by updating the plugin to
> 
> `"@braze/expo-plugin": "git+ssh://git@github.com:braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",`
> 
> but I still got the error when executing `expo rebuild`
> 
> ![Screenshot 2024-06-11 at 16 01 35](https://private-user-images.githubusercontent.com/14238145/338638690-bec7d8a6-8230-4988-ae7b-6a956106b3e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMTk3OTAsIm5iZiI6MTcxODExOTQ5MCwicGF0aCI6Ii8xNDIzODE0NS8zMzg2Mzg2OTAtYmVjN2Q4YTYtODIzMC00OTg4LWFlN2ItNmE5NTYxMDZiM2UyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDE1MjQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhYzYwOWVjZDI5NGNlMTA3Y2E4OWM2MTNjZGJlMmUwMmIyNDUzOTZjMWM3YWI2NDlkYmVjNGY0YThiYzY1OTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._UNBEZ2M3XTK8kpCnWnL7i3k7AKXfhTFw_EeHynm0Ts)
> 
> The build folder does not seem to have been generated. ![Screenshot 2024-06-11 at 16 22 14](https://private-user-images.githubusercontent.com/14238145/338647193-88e3fb16-f789-4501-9570-a4e5d757f9b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMTk3OTAsIm5iZiI6MTcxODExOTQ5MCwicGF0aCI6Ii8xNDIzODE0NS8zMzg2NDcxOTMtODhlM2ZiMTYtZjc4OS00NTAxLTk1NzAtYTRlNWQ3NTdmOWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDE1MjQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiMWViZWU5YzA1YThhYzQ4MTQxYjJiY2FkNzU0NzhlNGU1ZmFkY2JhZWM1OWNiNGJmZDdiZGVhODQ5M2U2MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rj73WKELjhPK1K37PRV6_itQtZXdpuVApV3hLwYInSg)
> 
> Could you please help? Thank you.

This is happening because there are no compiled binaries. 
What i had to do was pull down the code and host it in my repo, then add this to the scripts in package.json:
`"prepare": "expo-module prepare"`
it will compile when you npm install.

**Comment by YentingChen** [comment]
> > Hi @jerielng I tried to point to the repository by updating the plugin to
> > `"@braze/expo-plugin": "git+ssh://git@github.com:braze-inc/braze-expo-plugin-eas-preview.git#2.1.2",`
> > but I still got the error when executing `expo rebuild`
> > ![Screenshot 2024-06-11 at 16 01 35](https://private-user-images.githubusercontent.com/14238145/338638690-bec7d8a6-8230-4988-ae7b-6a956106b3e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMTk3OTAsIm5iZiI6MTcxODExOTQ5MCwicGF0aCI6Ii8xNDIzODE0NS8zMzg2Mzg2OTAtYmVjN2Q4YTYtODIzMC00OTg4LWFlN2ItNmE5NTYxMDZiM2UyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDE1MjQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhYzYwOWVjZDI5NGNlMTA3Y2E4OWM2MTNjZGJlMmUwMmIyNDUzOTZjMWM3YWI2NDlkYmVjNGY0YThiYzY1OTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._UNBEZ2M3XTK8kpCnWnL7i3k7AKXfhTFw_EeHynm0Ts)
> > The build folder does not seem to have been generated. ![Screenshot 2024-06-11 at 16 22 14](https://private-user-images.githubusercontent.com/14238145/338647193-88e3fb16-f789-4501-9570-a4e5d757f9b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMTk3OTAsIm5iZiI6MTcxODExOTQ5MCwicGF0aCI6Ii8xNDIzODE0NS8zMzg2NDcxOTMtODhlM2ZiMTYtZjc4OS00NTAxLTk1NzAtYTRlNWQ3NTdmOWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDE1MjQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiMWViZWU5YzA1YThhYzQ4MTQxYjJiY2FkNzU0NzhlNGU1ZmFkY2JhZWM1OWNiNGJmZDdiZGVhODQ5M2U2MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rj73WKELjhPK1K37PRV6_itQtZXdpuVApV3hLwYInSg)
> > Could you please help? Thank you.
> 
> @YentingChen This is happening because there are no compiled binaries. What i had to do was pull down the code and host it in my repo, then add this to the scripts in package.json: `"prepare": "expo-module prepare"` it will compile when you npm install.

Hi @krazykriskomar 

Thank you for the suggestion, which helps.
Can you share your methods of using EAS build? Thank you.
I got the error only through the EAS build, but the locally build works fine.

<img width="989" alt="Screenshot 2024-06-12 at 17 39 13" src="https://github.com/braze-inc/braze-expo-plugin/assets/14238145/bb0d5012-5c4b-4c9c-bf8f-9e5cc2ded547">

**Comment by krazykriskomar** [comment]
@YentingChen Ah yes, since EAS build does a npm ci and not an npm install, it won't do anything extra, like install does. The way I solved it was to download the repo and add it as an Expo plugin. 
![Xnip2024-06-13_08-43-47](https://github.com/braze-inc/braze-expo-plugin/assets/87501600/df42a98b-e039-4e8d-abed-468c626273a1)

**Comment by jerielng** [comment]
Hi all, upon closer review, it looks like there are no additional configurations/changes needed to successfully code-sign when building with EAS. This means that the `appleTeamId` configuration that was in the preview repo should not be necessary. To get this working end-to-end, you'll need to ensure that the bundle IDs for each app extension is added to your project configuration.

The currently released version should have everything needed to successfully build on EAS, but we've published some sample app updates in version 2.1.2 for further guidance and will have some updated documentation out shortly, which will live [here](https://www.braze.com/docs/developer_guide/platform_integration_guides/react_native/push_notifications) when it's published.

I'll close out this issue, but please feel free to re-open it if you face any additional issues. Thanks!
